tests/src/Functional/SettingsTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\collabora_online\Functional;
+
+use Drupal\Core\Url;
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Tests the Collabora configuration.
+ */
+class SettingsTest extends BrowserTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'collabora_online',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * Tests the configuration for the Collabora settings form.
+   */
+  public function testSettingsForm(): void {
+    $assert_session = $this->assertSession();
+
+    // User without permission can't access the configuration page.
+    $this->drupalGet(Url::fromRoute('collabora-online.settings'));
+    $assert_session->pageTextContains('You are not authorized to access this page.');
+    $assert_session->statusCodeEquals(403);
+
+    // User with permission can access the configuration page.
+    $user = $this->createUser(['administer site configuration']);
+    $this->drupalLogin($user);
+    $this->drupalGet(Url::fromRoute('collabora-online.settings'));
+    $assert_session->statusCodeEquals(200);
+
+    // Check fields and default values.
```

Comment:
I think the default values are coming from `config/install/collabora_online.settings.yml`.

Perhaps we should also test what we get if we delete all the settings. This would verify that the form does not break or produce notices if the settings are empty. Perhaps this could be done at the end of the test method.